controllers/contentlibrary/clustercontentlibraryitem/clustercontentlibraryitem_controller.go

```diff
@@ -159,63 +154,82 @@ func (r *Reconciler) ReconcileNormal(ctx goctx.Context, cclItem *imgregv1a1.Clus
 			savedStatus = cvmi.Status.DeepCopy()
 		}()
 
-		if utils.CheckItemReadyCondition(cclItem.Status.Conditions) {
-			conditions.MarkTrue(cvmi, vmopv1.VirtualMachineImageProviderReadyCondition)
-		} else {
+		ctx.CVMI = cvmi
+		if err := r.setUpCVMIFromCCLItem(ctx); err != nil {
+			ctx.Logger.Error(err, "Failed to set up ClusterVirtualMachineImage from ClusterContentLibraryItem")
+			return err
+		}
+
+		// Check if the item is ready.
+		if !utils.CheckItemReadyCondition(ctx.CCLItem.Status.Conditions) {
 			conditions.MarkFalse(cvmi,
 				vmopv1.VirtualMachineImageProviderReadyCondition,
 				vmopv1.VirtualMachineImageProviderNotReadyReason,
 				vmopv1.ConditionSeverityError,
 				"Provider item is not in ready condition",
 			)
-			logger.Info("ClusterContentLibraryItem is not ready yet, skipping further reconciliation")
+			ctx.Logger.Info("ClusterContentLibraryItem is not ready yet, skipping further reconciliation")
 			return nil
 		}
+		conditions.MarkTrue(cvmi, vmopv1.VirtualMachineImageProviderReadyCondition)
 
-		if err := r.setUpCVMIFromCCLItem(cvmi, cclItem); err != nil {
-			logger.Error(err, "Failed to set up ClusterVirtualMachineImage from ClusterContentLibraryItem")
-			return err
+		// Check if the item is security compliant.
+		cclItemSecurityCompliance := ctx.CCLItem.Status.SecurityCompliance
```

Comment:
Can this change on an existing item? Should the `syncImageContent` be done so the rest of the image status is updated if there are other item changes?

Reply:
Yep. Moved it inside the `setUpCVMIFromCCLItem` function to make things clearer here.